public/js/graph/react_graph.js

```diff
@@ -927,7 +989,16 @@ var Edge = React.createClass({
         }
     },
 
+    componentDidUpdate : function() {
+    	if (this.state.status === 'missing') {
```

Comment:
Make this more general: call `this.props.updateEdgeStatus` if `this.state.status` is different from the *old* state (add two parameters to the function - see documentation for `componentDidUpdate`